### ISSUE TITLE
Bug: [@typescript-eslint/consistent-type-imports] False positive and bad fix for "interface" is used by decoretor metadata

### ISSUE THREAD

**Issue by OpportunityLiu** [typescript-eslint/typescript-eslint#7214]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

When an interface is imported by
```ts
import type { Foo } from "./foo-bar.js"; // interface Foo {}
```
And used as constructor parameter type
```ts
export class B {
  constructor(readonly foo: Foo) {}
}
```
This rule will report 
```
Type import "Foo" is used by decorator metadata. eslint[@typescript-eslint/consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports)
```
And fix it to:
```ts
import { Foo } from "./foo-bar.js"; 
```
Which cause TypeScript to report:
```
'Foo' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled. ts(1484)
```

In fact, in this situation, even if a class is imported, it is still not used by decorator metadata, since TypeScript only includes experimental support for emitting certain types of metadata for declarations that have decorators. https://www.typescriptlang.org/docs/handbook/decorators.html#metadata
 
For example,
```ts
// no-metadata.ts
import { Bar } from "./foo-bar.js"; // class Bar {}

export class B {
  constructor(readonly bar: Bar) {}
}
```
will be transpiled to:
```js
// no-metadata.js
import { Bar } from "./foo-bar.js";
export class B {
    bar;
    constructor(bar) {
        this.bar = bar;
    }
}
```

If you use `import type { Bar } from "./foo-bar.js"` in this situation, `@typescript-eslint/consistent-type-imports` will also report it.

### Reproduction Repository Link

https://github.com/OpportunityLiu/consistent-type-imports-repro

### Repro Steps

1) clone the repo
2) `yarn install`
3) `yarn eslint interface-meta.ts`
3) `yarn eslint no-metadata.ts`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.0.0` |
| `@typescript-eslint/parser`            | `6.0.0` |
| `@typescript-eslint/scope-manager`     | `6.0.0` |
| `@typescript-eslint/typescript-estree` | `6.0.0` |
| `@typescript-eslint/type-utils`        | `6.0.0` |
| `@typescript-eslint/utils`             | `6.0.0` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.44.0` |
| `node`                                 | `18.16.1` |

**Comment by OpportunityLiu** [comment]
Downgraded to 5.62, it seems worked as expected

**Comment by bradzacher** [comment]
Duplicate of #5468
